.github/workflows/pr.yml

```diff
@@ -58,18 +57,44 @@ jobs:
         # 64 bit int
         cmake ${SHARED_OPT} -DBUILD_INDEX64=on -B lapack/build64
         cmake --build lapack/build64 ${PARALLEL} --target install
+    - name: Test if the changes affect this domain
+      id: test_domain
+      uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+      with:
+        script: |
+          function matchesPattern(filePaths) {
+            console.log('files:', filePaths)
+            // These directories contain domain specific code
+            const dirs = '(tests/unit_tests|examples|src|include/oneapi/mkl)'
+            const domains = '(blas|lapack|rng|dft)'
+            const domain = '${{ matrix.domain }}'
+            // matches changes to the domain of interest or non domain-specific code
```

Comment:
I don't fully understand the syntax here so just double-checking: does this mean that if there is a non domain-specific change, every domain will be tested?

Reply:
> does this mean that if there is a non domain-specific change, every domain will be tested?

Yes.

```
${dirs}/${domain}
```
Is the domain-specific code that should trigger a test. e.g. tests/unit_tests/blas
```
?!${dirs}/${domains}
```
Is all the non-domain specific code. The `?!` is a negation and `${dirs}/${domains}` is all the domain specific code for all domains. Testing is triggered if either of above matches.

Javascript is the underlying language for the GitHub Actions, and using a javascript regular expression makes it easy to test the code outside of the CI system. Simple whitelisting is likely to be wrong or out of date.

Reply:
Thank you, this looks good to me.